add_tests_for_main

myMordheim opened the pull request:
Unit-test for main.py

myMordheim committed:
add_tests_for_main

Unit-test for main.py

myMordheim committed:
Create .gitignore

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        collector = BooksCollector()
+        collector.add_new_book('LotR: Fellowship of the Python')
+        collector.set_book_genre('LotR: Fellowship of the Python', genre)
+        assert collector.books_genre['LotR: Fellowship of the Python'] == genre
```
Нужно исправить: здесь и далее: при наличии методов работы с атрибутом у тестируемой программы, необходимо использовать их, 
нам важно знать что не только атрибут изменился но и метод работы с ним работает, так и тестов будет меньше и атомарность не пострадает

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        collector.set_book_genre('LotR: Fellowship of the Python', 'Стендап')
+        assert collector.books_genre['LotR: Fellowship of the Python'] == ''
+
+    def test_get_book_genre_check_correct_method_work(self):
+        collector = BooksCollector()
+        collector.add_new_book('LotR: Fellowship of the Python')
+        collector.set_book_genre('LotR: Fellowship of the Python', 'Фантастика')
+        assert collector.get_book_genre('LotR: Fellowship of the Python') == 'Фантастика'
```
Этот тест покрывает верхние два

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        collector.set_book_genre('LotR: Return of the Snake King', 'Ужасы')
+        assert collector.get_books_for_children()
+
+    def test_add_book_in_favorites_add_book_in_favorites(self):
```
Тест - копия  test_get_list_of_favorites_books_check_correct_method_work

AlexKitIt commented on tests.py:
```diff
@@ -1,24 +1,85 @@
+        assert ['LotR: Return of the Snake King',
+                'LotR: Return of the Python King'] == collector.get_books_with_specific_genre('Ужасы')
+
+    def test_get_books_genre_check_correct_method_work(self):
+        collector = BooksCollector()
+        collector.add_new_book('LotR: Fellowship of the Python')
+        collector.set_book_genre('LotR: Fellowship of the Python', 'Фантастика')
+        assert collector.get_books_genre()
```
Метод уже проверен в test_add_new_book_add_one_book

myMordheim committed:
Add new tests, remove bad tests

Добавил тесты с новыми проверками, поменял ассерты на корректные, убрал дублирующиеся тесты
